jormungandr/src/secure/enclave.rs

```diff
@@ -4,13 +4,14 @@ use crate::blockcfg::{
 };
 use chain_impl_mockchain::leadership::{Leader, LeaderOutput, Leadership};
 use jormungandr_lib::interfaces::EnclaveLeaderId as LeaderId;
-use std::collections::BTreeMap;
+use std::collections::{BTreeMap, HashMap, HashSet};
 use std::sync::Arc;
 use tokio02::sync::RwLock;
 
 #[derive(Clone)]
 pub struct Enclave {
     leaders: Arc<RwLock<BTreeMap<LeaderId, Leader>>>,
+    added_cache: Arc<RwLock<HashMap<String, LeaderId>>>,
```

Comment:
I used `String` but maybe we could use `PublicKey` instead, I wanted to go ahead and used the easy way first but if `PublicKey` is hashable should be fine also.